source/SH3/graphics/texture.cpp

```diff
@@ -13,48 +13,330 @@
 #include <SH3/arc/vfile.hpp>
 
 #include <cassert>
+#include <fstream>
 
+#define HEADER_START 64
+
+static std::size_t offset = 0;
 
 using namespace sh3_graphics;
 
+#pragma pack(push, 1)
+struct rgba
+{
+    std::uint8_t r, g, b, a;
+};
+
+struct rgba16
+{
+    uint8_t r : 5;
+    uint8_t g : 5;
+    uint8_t b : 5;
+    uint8_t a : 1;
+};
+
+struct rgb
+{
+    std::uint8_t r, g, b;
+};
+
+struct bgra
+{
+    std::uint8_t b, g, r, a;
+};
+
+struct bgr
+{
+    std::uint8_t b, g, r;
+};
+
+struct tga_header
+{
+    std::uint8_t id_size;   // Size of the ID field that follows this header (usually 0)
+    std::uint8_t colormap;  // Is this image paletted (THIS MUST ALWAYS BE 0 FOR US!)
+    std::uint8_t type;      // Type of image (ALWAYS 2 FOR RGB24!)
+    std::uint8_t unused[5]; // Colormap crap
+
+    std::uint16_t x_origin;  // Co-ordinate for first X value
+    std::uint16_t y_origin;  // Co-ordinate for first Y value
+    std::uint16_t width;
+    std::uint16_t height;
+    std::uint8_t bpp;       // Bits per pixel
+    std::uint8_t flags;
+};
+
+#pragma pack(pop)
+
+/**
+ *  Dump a texture to a Microsoft BMP/DIB
+ *
+ *  @param width - The Width of this texture.
+ *  @param height - The Height of this texture.
+ *  @param data - Pixel data
+ */
+void Dump2Bitmap(std::uint32_t width, std::uint32_t height, std::vector<std::uint8_t>& data)
```

Comment:
We'll want to rename it to `DumpRGB2Bitmap` at least, to make sure it's not used for RGBA.
Also this function can be an an anonymous namespace.